lib/datasource/cdnjs/common.ts

```diff
@@ -0,0 +1,33 @@
+import { fold } from 'fp-ts/Either';
+import { identity, pipe } from 'fp-ts/function';
+import * as t from 'io-ts';
+import type { CdnjsResponse } from './types';
+
+const Response = t.intersection([
+  t.type({
+    assets: t.array(
+      t.type({
+        version: t.string,
+        files: t.array(t.string),
+        sri: t.record(t.string, t.string),
+      })
+    ),
+  }),
+  t.partial({
+    homepage: t.string,
+    repository: t.partial({ url: t.string }),
+  }),
+]);
+
+export type IResponse = CdnjsResponse;
+
+export function validResponse(input: unknown): IResponse {
+  const res: IResponse | null = pipe(
+    Response.decode(input),
+    fold(() => null, identity)
+  );
+  if (!res) {
+    throw new Error('Cdnjs: invalid response');
```

Comment:
`io-ts` is well-crafted functional library, but we are likely to neglect this purity like I did here in favor of convenience